README.md
# Button Animations

![dart](https://img.shields.io/badge/dart-2.19.5-blue?logo=dart&logoColor=blue&labelColor=black&color=0175C2) ![flutter](https://img.shields.io/badge/flutter-3.7.8-blue?logo=flutter&logoColor=blue&labelColor=black) [![pub package](https://img.shields.io/pub/v/path_provider.svg)](https://pub.dev/packages/button_animations)
<img src="https://visitor-badge.glitch.me/badge?page_id=anshrathod.button_animations">

The [button_animations](https://pub.dev/packages/button_animations) is a package to make highly customizable [Flutter](https://flutter.dev/) widgets with 3d animations.

## Getting Started

To use this plugin, add the latest version of `button_animations` as a [dependency in your pubspec.yaml file](https://flutter.io/platform-plugins/).

### Platform Support

| Android | iOS | Web | MacOS | Linux | Windows |
| :-----: | :-: | :---: | :-----: | :-: | :---: |
| &#9989;  | &#9989; | &#9989; | &#9989;  | &#9989; | &#9989; |

### Screenshots

<p>
<img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/3.jpg" width="30%" height="424">
<img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/6.jpg" width="30%" height="424">
<img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/7.jpg" width="30%" height="424">
</p>

### Parameters and their Description

|      Datatype      |     Parameter      |                                                        Description                                                         |      Default Value       | Required |
| :----------------: | :----------------: | :------------------------------------------------------------------------------------------------------------------------: | :----------------------: | :------: |
|    `function `     |       onTap        |                                 The function that is called whenever the widget is tapped                                  |            --            |   Yes    |
|      `widget`      |       child        |                                                      The child widget                                                      |            --            |   Yes    |
| `PredefinedThemes` |        type        |      There are 16 predefined themes that you can choose from if you dont want to customize the widget yourself       | PredefinedThemes.primary |    No    |
|      `Curve`       |   animationCurve   |                                          The curve that the animation will follow                                          |      Curves.easeIn       |    No    |
|       `bool`       |      enabled       |                                       To check whether the button is enabled or not                                       |           true           |    No    |
|      `bool`       |    isMultiColor    |                               To check whether the button has multiple color gradient or not                               |          false           |    No    |
|       `bool`       |     isOutline      |                                      To check whether the button has an outline or not                                      |          false           |    No    |
|      `bool `       |     darkshadow     |                                    To check whether the button has dark or light shadow                                    |           true           |    No    |
|       `int`        |      duration      |                                     The time that the animation takes in milliseconds                                      |            70            |    No    |
|      `double`      |       height       |                                                  The height of the widget                                                  |            64            |    No    |
|      `double`      |       width        |                                                  The width of the widget                                                   |           200            |    No    |
|     `double `      |     blurRadius     |                                        The radius of the blur effect of the widget                                         |            0             |    No    |
|     `double `      |    borderRadius    |                                          The radius of the borders of the widget                                           |            12            |    No    |
|     `double `      | shadowHeightBottom |                The height of the shadow and the animation of the widget from the bottom of the child widget                |            4             |    No    |
|     `double `      |  shadowHeightLeft  |                 The height of the shadow and the animation of the widget from the left of the child widget                 |            0             |    No    |
|      `double`      |    borderWidth     |                                           The width of the border of the widget                                            |            1             |    No    |
|      `Color`       |    borderColor     | The color of the border of the widget (if type is not null, this will not work & isOutline should be true for this to work) |       Colors.black       |    No    |
|      `Color`       |     blurColor      |                        The color of the blur of the widget(if type is not null, this will not work)                        |       Colors.black       |    No    |
|      `Color`       |       color        |                              The color of the widget(if type is not null, this will not work)                              |       Colors.blue        |    No    |
|      `Color`       |    shadowColor     |                       The color of the shadow of the widget(if type is not null, this will not work)                       |       Colors.black       |    No    |
|   `List<Color>`    |       colors       |     The list of colors for the gradient for the background of the widget(isMulticolor should be true for this to work)     |            []            |    No    |

### PredefinedThemes

|    Theme Name    |                                                           Example                                                           |
| :--------------: | :-------------------------------------------------------------------------------------------------------------------------: |
|     success      |   <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/success.png" width="300">    |
|  successOutline  |  <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/successout.png" width="300">  |
|      danger      |    <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/danger.png" width="300">    |
|  dangerOutline   |  <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/dangerout.png" width="300">   |
|     warning      |   <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/warning.png" width="300">    |
|  warningOutline  |  <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/warningout.png" width="300">  |
|       info       |     <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/info.png" width="300">     |
|   infoOutline    |   <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/infoout.png" width="300">    |
|     primary      |   <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/primary.jpg" width="300">    |
|  primaryOutline  |  <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/primaryout.jpg" width="300">  |
|    secondary     |  <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/secondary.jpg" width="300">   |
| secondaryOutline | <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/secondaryout.png" width="300"> |
|       dark       |     <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/dark.jpg" width="300">     |
|   darkOutline    |   <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/darkout.png" width="300">    |
|      light       |    <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/light.png" width="300">     |

## Example

#### For any of the predefined themes you just have to add the child widget, type, and the onTap function.

##### 1. Add borderRadius and, isOutline to add more customization.

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/danger.png" width="500">

```ruby
AnimatedButton(
  child: Text(
    'Danger', // add your text here
    style: TextStyle(
      color: Colors.white,
    ),
  ),
  type: PredefinedTheme.danger,
  isOutline: true,
  borderWidth: 1,
  onTap: () {},
),
AnimatedButton(
  child: Text(
    'Danger', // add your text here
    style: TextStyle(
      color: Colors.white,
    ),
  ),
  type: PredefinedTheme.danger,
  darkshadow: false,
  isOutline: true,
  borderWidth: 1,

  onTap: () {},
),
```

##### 2. Use blurRadius and blurColor

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/dark.jpg" width="500">

```ruby
AnimatedButton(
  child: Text(
    'Dark', // add your text here
    style: TextStyle(
      color: Colors.white,
    ),
  ),
  type: PredefinedTheme.dark,
  blurRadius: 10,
  onTap: () {},
),
AnimatedButton(
  child: Text(
    'Danger', // add your text here
    style: TextStyle(
      color: Colors.white,
    ),
  ),
  type: PredefinedTheme.danger,
  darkshadow: false,
  blurRadius: 10,
  onTap: () {},
),
```

##### 3. Use all other parameters to make custom buttons

- Gradient Buttons

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/gradient.jpg" width="500">

```ruby
Row(
  children: [
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Text("Gradient Button"),
        onTap: () {},
        isMultiColor: true,
        colors: [
          Colors.red[100],
          Colors.blue[200],
        ],
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Text("Gradient Button with outline"),
        onTap: () {},
        isMultiColor: true,
        isOutline: true,
        colors: [
          Colors.red[100],
          Colors.blue[200],
        ],
      ),
    ),
  ],
),
Row(
  children: [
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Text(
          "Gradient Button",
          style: TextStyle(
            color: Colors.white,
          ),
        ),
        onTap: () {},
        isMultiColor: true,
        colors: [
          Colors.green,
          Colors.blue,
        ],
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Text(
          "Gradient Button with outline",
          style: TextStyle(
            color: Colors.white,
          ),
        ),
        onTap: () {},
        isMultiColor: true,
        isOutline: true,
        colors: [
          Colors.green,
          Colors.blue,
        ],
      ),
    ),
  ],
),

```

- Rounded Buttons

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/rounded.jpg" width="500">

```ruby
Row(
  children: [
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Text("Rounded Button"),
        onTap: () {},
        height: 60,
        type: null,
        borderRadius: 30,
        color: Colors.cyan,
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Text("Rounded Button with outline"),
        onTap: () {},
        height: 60,
        type: null,
        borderRadius: 30,
        isOutline: true,
        color: Colors.amber,
      ),
    ),
  ],
),

```

- Rounded Buttons with Icons

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/roundedicon.jpg" width="500">

```ruby
Row(
  children: [
    Padding(
      padding: const EdgeInsets.all(12.0),
      child: AnimatedButton(
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
          children: [
            Icon(Icons.home),
            Text("Button with Icon"),
          ],
        ),
        onTap: () {},
        type: null,
        height: 60,
        borderRadius: 30,
        color: Colors.teal,
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(2.0),
      child: AnimatedButton(
        child: Row(
          mainAxisAlignment: MainAxisAlignment.spaceEvenly,
          children: [
            Icon(Icons.home, color: Colors.white),
            Text("Button with Icon & outline",
                style: TextStyle(color: Colors.white)),
          ],
        ),
        onTap: () {},
        type: null,
        height: 60,
        width: 220,
        borderRadius: 30,
        isOutline: true,
        color: Colors.deepPurple,
      ),
    ),
  ],
),

```

- Social Media Buttons

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/socialmedia.jpg" width="500">

```ruby
Row(
  mainAxisAlignment: MainAxisAlignment.spaceEvenly,
  children: [
    Padding(
      padding: const EdgeInsets.all(2),
      child: AnimatedButton(
        child: Icon(
          FontAwesomeIcons.facebookF,
          color: Colors.white,
          size: 25,
        ),
        onTap: () {},
        type: null,
        height: 45,
        width: 45,
        borderRadius: 22.5,
        color: Color(0xFF49659F),
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(2),
      child: AnimatedButton(
        child: Icon(
          FontAwesomeIcons.twitter,
          color: Colors.white,
          size: 25,
        ),
        onTap: () {},
        type: null,
        height: 45,
        width: 45,
        borderRadius: 22.5,
        color: Color(0xFF1DA1F2),
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(2),
      child: AnimatedButton(
        child: Icon(
          FontAwesomeIcons.google,
          color: Colors.white,
          size: 25,
        ),
        onTap: () {},
        type: null,
        height: 45,
        width: 45,
        borderRadius: 22.5,
        color: Colors.blue,
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(2),
      child: AnimatedButton(
        child: Icon(
          FontAwesomeIcons.microsoft,
          color: Colors.lightBlue,
          size: 25,
        ),
        onTap: () {},
        type: null,
        height: 45,
        width: 45,
        borderRadius: 22.5,
        color: Colors.white,
      ),
    ),
    Padding(
      padding: const EdgeInsets.all(2),
      child: AnimatedButton(
        child: Icon(
          FontAwesomeIcons.instagram,
          color: Colors.white,
          size: 25,
        ),
        onTap: () {},
        type: null,
        height: 45,
        width: 45,
        borderRadius: 22.5,
        color: Color(0xFFB81877),
      ),
    ),
  ],
),

```

- Sign in Buttons

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/signin.jpg" width="300">

```ruby
Padding(
  padding: const EdgeInsets.all(10),
  child: AnimatedButton(
    child: Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: [
        Icon(
          FontAwesomeIcons.linkedinIn,
          color: Colors.white,
          size: 25,
        ),
        Text(
          "Sign in with LinkedIn",
          style: TextStyle(
            color: Colors.white,
            fontSize: 20.0,
          ),
        )
      ],
    ),
    onTap: () {},
    type: null,
    height: 50,
    shadowHeightBottom: 4,
    shadowHeightLeft: 4,
    width: 275,
    borderRadius: 25,
    color: Color(0xFF0077B5),
  ),
),
Padding(
  padding: const EdgeInsets.all(10),
  child: AnimatedButton(
    child: Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: [
        Icon(
          FontAwesomeIcons.pinterestP,
          color: Colors.white,
          size: 25,
        ),
        Text(
          "Sign in with Pinterest",
          style: TextStyle(
            color: Colors.white,
            fontSize: 20.0,
          ),
        )
      ],
    ),
    onTap: () {},
    type: null,
    height: 50,
    isOutline: true,
    shadowHeightBottom: 4,
    shadowHeightLeft: 4,
    width: 250,
    borderRadius: 2,
    color: Colors.red,
  ),
),
Padding(
  padding: const EdgeInsets.all(10),
  child: AnimatedButton(
    child: Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: [
        Icon(
          FontAwesomeIcons.githubAlt,
          color: Colors.white,
          size: 25,
        ),
        Text(
          "Sign in with GitHub",
          style: TextStyle(
            color: Colors.white,
            fontSize: 20.0,
          ),
        )
      ],
    ),
    onTap: () {},
    type: null,
    height: 50,
    shadowHeightBottom: 4,
    shadowHeightLeft: 4,
    width: 250,
    shadowColor: Colors.grey,
    blurColor: Colors.grey,
    blurRadius: 5,
    borderRadius: 10,
    color: Colors.black,
  ),
),
Padding(
  padding: const EdgeInsets.all(10),
  child: AnimatedButton(
    child: Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: [
        Icon(
          FontAwesomeIcons.apple,
          color: Colors.white,
          size: 25,
        ),
        Text(
          "Sign in with Apple",
          style: TextStyle(
            color: Colors.white,
            fontSize: 20.0,
          ),
        )
      ],
    ),
    onTap: () {},
    type: null,
    shadowColor: Colors.pink,
    height: 50,
    width: 250,
    shadowHeightBottom: 6,
    shadowHeightLeft: 4,
    borderRadius: 10,
    color: Color(0xFF333333),
  ),
),
Padding(
  padding: const EdgeInsets.all(10),
  child: AnimatedButton(
    child: Row(
      mainAxisAlignment: MainAxisAlignment.spaceAround,
      children: [
        Icon(
          Icons.mail_outline,
          color: Colors.white,
          size: 25,
        ),
        Text(
          "Sign in with Email",
          style: TextStyle(
            color: Colors.white,
            fontSize: 20.0,
          ),
        )
      ],
    ),
    onTap: () {},
    type: null,
    height: 50,
    width: 250,
    isOutline: true,
    darkShadow: false,
    shadowHeightBottom: 4,
    shadowHeightLeft: 4,
    borderRadius: 10,
    color: Colors.lightBlueAccent,
  ),
),

```

- Custom Buttons

 <img src="https://raw.githubusercontent.com/anshrathod/button_animations/master/Screen-shots/imagebutton.jpg" width="250">

```ruby
Padding(
  padding: const EdgeInsets.all(10),
  child: AnimatedButton(
    child: ClipRRect(
      borderRadius: BorderRadius.circular(10.0),
      child: Image.network(
        "my_logo_image_url",
        fit: BoxFit.fill,
      ),
    ),
    onTap: () {},
    type: null,
    height: 144,
    width: 144,
    isOutline: true,
    shadowHeightBottom: 4,
    shadowHeightLeft: 4,
    borderRadius: 10,
    color: Color(0xFF033249),
  ),
),

```


CHANGELOG.md
## 2.0.0+3
- Added null safety checks
- Made package comptible with dart 3
## 1.0.1+2
- Cleaned example and added comments to the package.
## 1.0.1
- Added xample.
## 1.0.0
- Initial open source release.


test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:button_animations/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


example/test/widget_test.dart
// This is a basic Flutter widget test.
//
// To perform an interaction with a widget in your test, use the WidgetTester
// utility that Flutter provides. For example, you can send tap and scroll
// gestures. You can also use WidgetTester to find child widgets in the widget
// tree, read text, and verify that the values of widget properties are correct.

import 'package:flutter/material.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:exampl/main.dart';

void main() {
  testWidgets('Counter increments smoke test', (WidgetTester tester) async {
    // Build our app and trigger a frame.
    await tester.pumpWidget(MyApp());

    // Verify that our counter starts at 0.
    expect(find.text('0'), findsOneWidget);
    expect(find.text('1'), findsNothing);

    // Tap the '+' icon and trigger a frame.
    await tester.tap(find.byIcon(Icons.add));
    await tester.pump();

    // Verify that our counter has incremented.
    expect(find.text('0'), findsNothing);
    expect(find.text('1'), findsOneWidget);
  });
}


lib/constants.dart
import 'package:flutter/material.dart';

enum PredefinedThemes {
  success,
  successOutline,
  danger,
  dangerOutline,
  warning,
  warningOutline,
  info,
  infoOutline,
  primary,
  primaryOutline,
  secondary,
  secondaryOutline,
  dark,
  darkOutline,
  light,
}

List definedColors = [
  // success
  {
    "color": const Color(0xFF5CB85C),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // successOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFF5CB85C),
    "shadowColor": const Color(0xFF5CB85C),
    "blurColor": const Color(0xFF5CB85C)
  },
  // danger
  {
    "color": const Color(0xFFD9534F),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // dangerOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFFD9534F),
    "shadowColor": const Color(0xFFD9534F),
    "blurColor": const Color(0xFFD9534F)
  },
  // warning
  {
    "color": const Color(0xFFF0AD4E),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // warningOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFFF0AD4E),
    "shadowColor": const Color(0xFFF0AD4E),
    "blurColor": const Color(0xFFF0AD4E)
  },
  // info
  {
    "color": const Color(0xFF5BC0DE),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // infoOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFF5BC0DE),
    "shadowColor": const Color(0xFF5BC0DE),
    "blurColor": const Color(0xFF5BC0DE)
  },
  // primary
  {
    "color": const Color(0xFF0275D8),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // primaryOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFF0275D8),
    "shadowColor": const Color(0xFF0275D8),
    "blurColor": const Color(0xFF0275D8)
  },
  // secondary
  {
    "color": const Color(0xFF6B5B95),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // secondaryOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFF6B5B95),
    "shadowColor": const Color(0xFF6B5B95),
    "blurColor": const Color(0xFF6B5B95)
  },
  // dark
  {
    "color": const Color(0xFF292B2C),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  // darkOutline
  {
    "color": const Color(0xFFFAFAFA),
    "borderColor": const Color(0xFF292B2C),
    "shadowColor": const Color(0xFF292B2C),
    "blurColor": const Color(0xFF292B2C)
  },
  {
    "color": const Color(0xFFF7F7F7),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  },
  {
    "color": const Color(0xFFBFBFC4),
    "borderColor": Colors.black,
    "shadowColor": Colors.black,
    "blurColor": Colors.black
  }
];


example/lib/main.dart
import 'package:button_animations/button_animations.dart';
import 'package:button_animations/constants.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(MyApp());
}

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  MyHomePage({Key key, this.title}) : super(key: key);

  final String title;

  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("Button Animations"),
      ),
      body: Padding(
        padding: const EdgeInsets.all(20.0),
        child: SingleChildScrollView(
          child: Column(
            children: [
              Row(
                children: [
                  Padding(
                    padding: const EdgeInsets.all(10),
                    child: AnimatedButton(
                        child: Text(
                          'Success',
                          style: TextStyle(
                            color: Colors.white,
                          ),
                        ),
                        type: PredefinedThemes.success,
                        onTap: () {}),
                  ),
                  Padding(
                    padding: const EdgeInsets.all(10),
                    child: AnimatedButton(
                        child: Text(
                          'Success',
                          style: TextStyle(
                            color: Colors.green,
                          ),
                        ),
                        type: PredefinedThemes.successOutline,
                        onTap: () {}),
                  ),
                ],
              ),
              SizedBox(height: 25),
              Row(
                children: [
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Text("Gradient Button"),
                      onTap: () {},
                      isMultiColor: true,
                      colors: [
                        Colors.red[100],
                        Colors.blue[200],
                      ],
                    ),
                  ),
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Text("Gradient Button with outline"),
                      onTap: () {},
                      isMultiColor: true,
                      isOutline: true,
                      colors: [
                        Colors.red[100],
                        Colors.blue[200],
                      ],
                    ),
                  ),
                ],
              ),
              Row(
                children: [
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Text(
                        "Gradient Button",
                        style: TextStyle(
                          color: Colors.white,
                        ),
                      ),
                      onTap: () {},
                      isMultiColor: true,
                      colors: [
                        Colors.green,
                        Colors.blue,
                      ],
                    ),
                  ),
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Text(
                        "Gradient Button with outline",
                        style: TextStyle(
                          color: Colors.white,
                        ),
                      ),
                      onTap: () {},
                      isMultiColor: true,
                      isOutline: true,
                      colors: [
                        Colors.green,
                        Colors.blue,
                      ],
                    ),
                  ),
                ],
              ),
              // SizedBox(height: 25),
              Row(
                children: [
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Text("Rounded Button"),
                      onTap: () {},
                      height: 60,
                      type: null,
                      borderRadius: 30,
                      color: Colors.cyan,
                    ),
                  ),
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Text("Rounded Button with outline"),
                      onTap: () {},
                      height: 60,
                      type: null,
                      borderRadius: 30,
                      isOutline: true,
                      color: Colors.amber,
                    ),
                  ),
                ],
              ),
              // SizedBox(height: 25),
              Row(
                children: [
                  Padding(
                    padding: const EdgeInsets.all(12.0),
                    child: AnimatedButton(
                      child: Row(
                        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                        children: [
                          Icon(Icons.home),
                          Text("Button with Icon"),
                        ],
                      ),
                      onTap: () {},
                      type: null,
                      height: 60,
                      borderRadius: 30,
                      color: Colors.teal,
                    ),
                  ),
                  Padding(
                    padding: const EdgeInsets.all(2.0),
                    child: AnimatedButton(
                      child: Row(
                        mainAxisAlignment: MainAxisAlignment.spaceEvenly,
                        children: [
                          Icon(Icons.home, color: Colors.white),
                          Text("Button with Icon & outline",
                              style: TextStyle(color: Colors.white)),
                        ],
                      ),
                      onTap: () {},
                      type: null,
                      height: 60,
                      width: 220,
                      borderRadius: 30,
                      isOutline: true,
                      color: Colors.deepPurple,
                    ),
                  ),
                ],
              ),
              Padding(
                padding: const EdgeInsets.all(10),
                child: AnimatedButton(
                  child: ClipRRect(
                    borderRadius: BorderRadius.circular(10.0),
                    child: Image.network(
                      "https://lh3.googleusercontent.com/baE_tms7XKv_EnCdJdTRsUSoLx7MId-lKcUH7T3tNDevG4FqpyHfF5h_zr_KdksCMQ=w150-h144-n-rw",
                      fit: BoxFit.fill,
                    ),
                  ),
                  onTap: () {},
                  type: null,
                  height: 144,
                  width: 144,
                  isOutline: true,
                  shadowHeightBottom: 4,
                  shadowHeightLeft: 4,
                  borderRadius: 10,
                  color: Color(0xFF033249),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
